### PR TITLE
add a patch for the case +host arch=ppc64le (fixes #22758)

### LINKED ISSUE #22758: phist@1.9.3 build fails for power9le: -march=native not recognized
`phist@1.9.3` fails to build optimized for `power9le` using `spack@develop` (d978d72e19574dc46dce84704349b716469babe2 from `Fri Apr 2 16:56:16 2021 +0200`)

Concrete spec: [phist.spec.yaml.txt](https://github.com/spack/spack/files/6250511/phist.spec.yaml.txt)
Full build log: [phist-build-out.txt](https://github.com/spack/spack/files/6250471/phist-build-out.txt)

```
$> spack install -f ./phist.spec.yaml
...
==> Installing phist-1.9.3-jxrqv52bq6ybd3j7rykmcoh25x4b57vi
==> Fetching https://spack-llnl-mirror.s3-us-west-2.amazonaws.com/_source-cache/archive/3a/3ab7157e9f535a4c8537846cb11b516271ef13f82d0f8ebb7f96626fb9ab86cf.tar.gz
######################################################################################################################################################################### 100.0%
==> Applied patch /opt/spack/var/spack/repos/builtin/packages/phist/sbang.patch
==> phist: Executing phase: 'cmake'
==> phist: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j32'
...
  >> 421    gfortran: error: unrecognized command line option '-march=native'; did you mean '-mcpu=native'?
  >> 422    make[2]: *** [fortran_bindings/CMakeFiles/phist_fort.dir/build.make:760: fortran_bindings/CMakeFiles/phist_fort.dir/phist_types.F90.o] Error 1
     423    make[2]: *** Waiting for unfinished jobs....
  >> 424    gfortran: error: unrecognized command line option '-march=native'; did you mean '-mcpu=native'?
  >> 425    make[2]: *** [fortran_bindings/CMakeFiles/phist_fort.dir/build.make:773: fortran_bindings/CMakeFiles/phist_fort.dir/phist_testing.F90.o] Error 1
     426    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-phist-1.9.3-jxrqv52bq6ybd3j7rykmcoh25x4b57vi/spack-build-jxrqv52'
  >> 427    make[1]: *** [CMakeFiles/Makefile2:3182: fortran_bindings/CMakeFiles/phist_fort.dir/all] Error 2
     428    make[1]: Leaving directory '/tmp/root/spack-stage/spack-stage-phist-1.9.3-jxrqv52bq6ybd3j7rykmcoh25x4b57vi/spack-build-jxrqv52'
     429    make: *** [Makefile:185: all] Error 2

See build log for details:
  /tmp/root/spack-stage/spack-stage-phist-1.9.3-jxrqv52bq6ybd3j7rykmcoh25x4b57vi/spack-build-out.txt
```

This was in an `Ubuntu 20.04 w/ GCC 9.3.0` environment, but I observe the same in `RHEL 7.6 w/ GCC 9.3.0`.

@jthies

### PR DESCRIPTION
**Pull request by jthies**

### CODE REVIEW

**Review by bernhardkaindl — DISMISSED** [review]

File: var/spack/repos/builtin/packages/phist/package.py [hunk 1]
```diff
@@ -102,8 +102,8 @@ class Phist(CMakePackage):
 
     # ###################### Patches ##########################
 
+    patch('ppc64_sse.patch', when='@1.7.4:1.9.4')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/phist/package.py:108]: Can you add a brief comment to explain why this is needed and maybe point to its provenance? A one liner is fine, it's just to have some clue in the package.py file.

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/phist/package.py (hunk 1 above)
Comment (bernhardkaindl) [reply var/spack/repos/builtin/packages/phist/package.py:108]: While at it, as PPC isn't the only architecture family without X86 SSE intrinsics:
 
Maybe it becomes more obvious (and also semantically correct), if the patch isn't like this:
```py
+if(PHIST_HOST_OPTIMIZE)
+  if (CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "ppc64le")
+    add_definitions(-DNO_WARN_X86_INTRINSICS)
+  endif()
+endif()
```
but rather like:
```py
+if(PHIST_HOST_OPTIMIZE)
+  if(NOT CMAKE_SYSTEM_PROCESSOR MATCHES "amd64.*|x86_64.*|AMD64.*")
+    add_definitions(-DNO_WARN_X86_INTRINSICS)
+  endif()
+endif()
```
This also makes sense as `aarch64` is also used with spack and maybe `RISCV` in the future.

Caution: Didn't test this check, it's based on internet search:
https://cmake.org/cmake/help/latest/variable/CMAKE_SYSTEM_PROCESSOR.html
https://cmake.org/cmake/help/latest/variable/CMAKE_HOST_SYSTEM_PROCESSOR.html
https://github.com/opencv/opencv/blob/1261f250c6cb33e411d257406e6ac718bed6fdc3/cmake/OpenCVDetectCXXCompiler.cmake#L87
With the `NOT` added in front (yes it's weird, but so is the CMAKE if NOT inversion AFAIK).

**Review by bernhardkaindl — DISMISSED** [review]

**Comment by bernhardkaindl** [comment]
@alalazo The comment you requested was added. It is ready to be approved(merge is enabled)

**Review by alalazo — DISMISSED** [review]

**Comment by alalazo** [comment]
@jthies Can you solve the conflict?

**Comment by bernhardkaindl** [comment]
@jthies Resolving the conflict should be easy. You could even do it via the button "Resolve conflicts" in the PR page here.

To conflicting commit was likely the removal of all the .99 version number extensions for ranges by haampie as they become obsolete after his commit making them oboslete.

**Review by alalazo — APPROVED** [review]

**Pull request merged**